stlab/concurrency/utility.hpp

```diff
@@ -108,6 +108,49 @@ T blocking_get(future<T> x) {
     return std::move(result.get());
 }
 
+template <typename T>
+boost::optional<T> blocking_get(future<T> x,const std::chrono::nanoseconds& timeout) {
+    boost::optional<T> result;
+    std::exception_ptr error = nullptr;
+
+    bool flag{false};
+    std::condition_variable condition;
+    std::mutex m;
+
+    auto hold = std::move(x).recover(immediate_executor, [&](auto&& r) {
+        if (r.error())
+            error = std::forward<decltype(r)>(r).error().value();
+        else
+            result = std::forward<decltype(r)>(r).get_try().value();
+
+        {
+            std::unique_lock<std::mutex> lock{m};
+            flag = true;
+            /*
+                WARNING : Calling `notify_one()` inside the lock is a pessimization because
+                it means the code waiting will block aquiring the lock as soon as it wakes up.
+
+                However, if we do the notificiation outside the lock, blocking_get will return
+                and we'll be calling a destructed condition variable.
+            */
+            condition.notify_one();
+        }
+    });
+    
+    std::unique_lock<std::mutex> lock{m};
+    while (!flag) {
+        if(condition.wait_for(lock, timeout) == std::cv_status::timeout) {
+            hold.reset();
+            return result;
+        }
+    }
+
+    if (error)
+        std::rethrow_exception(error);
+
+    return result;
+}
+
 inline void blocking_get(future<void> x) {
```

Comment:
Is the blocking_get(future<void>, timeout) missing on purpose?

Reply:
Never had one. Can add one though.

Reply:
@FelixPetriconi For the future<void> do you think it should be 

`inline bool blocking_get(future<void>x)`

With the bool indicating if it timed out or completed?